### ISSUE TITLE
Scoped instances not working in Query Handlers

### ISSUE THREAD

**Issue by frankebersoll** [eventflow/EventFlow#484]
**TL;DR:** QueryProcessor is singleton and uses caching, Query Handlers don't get scoped instances (e.g. DbContext with http request lifetime).

In #445 I added some tests to reproduce an issue where using a scoped instance from an `IQueryHandler` crashes: ASP.NET Core uses scopes for each request and `AddDbContext<TContext>()` registers the Entity Framework context as scoped instance, so it will exist once per request.
Because `IQueryProcessor` is a singleton by default, it uses the root `IResolver` instead of a scoped one. It even caches handler instances, so we can't really use scoped dependencies for query handlers for the time being.
While the ServiceProvider container does some checks for scoped access and throws whenever a scoped object tree trys to access the root resolver, Autofac does not, which might result in some hard to debug instance problems.

**Comment by frankebersoll** [comment]
Fixed in #488
